clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -47,6 +47,11 @@ public OutputStream wrapForOutput(ByteBufferOutputStream buffer, byte messageVer
         public InputStream wrapForInput(ByteBuffer buffer, byte messageVersion, BufferSupplier decompressionBufferSupplier) {
             return new ByteBufferInputStream(buffer);
         }
+
+        @Override
+        public int getRecommendedDOutSize() {
+            return 2 * 1024; // 2KB
```

Comment:
What's the meaning of this for an uncompressed stream?

Reply:
Changed to throw an UnSupportedException in the abstract class and Uncompressed case does not overload this method anymore.